test/pr.ts

```diff
@@ -23,60 +23,287 @@ before(() => {
-        .resolves(createRefResponse);
-      // tests
-      await openPullRequest(octokit, upstream, origin, description);
-      sinon.assert.calledOnceWithExactly(stub, {
-        owner: upstream.owner,
-        repo: origin.repo,
-        title: description.title,
-        head: 'origin-owner:pr-test-branch',
-        base: 'master',
-        body: description.body,
-        maintainer_can_modify: true,
-      });
-      // restore
-      stub.restore();
+  it('Invokes octokit pull create when there is not an existing pull request open', async () => {
+    // setup
+    const responseCreatePullData = await import(
+      './fixtures/create-pr-response.json'
+    );
+    const createPrResponse = {
+      headers: {},
+      status: 200,
+      url: 'http://fake-url.com',
+      data: responseCreatePullData,
+    };
+    const listPullResponse = {
+      headers: {},
+      status: 200,
+      url: 'http://fake-url.com',
+      data: [],
+    };
+    sandbox.stub(octokit.pulls, 'list').resolves(listPullResponse);
+    const stub = sandbox
+      .stub(octokit.pulls, 'create')
+      .resolves(createPrResponse);
+    // tests
+    await openPullRequest(octokit, upstream, origin, description);
+    sandbox.assert.calledOnceWithExactly(stub, {
+      owner: upstream.owner,
+      repo: origin.repo,
+      title: description.title,
+      head: 'origin-owner:pr-test-branch',
+      base: 'master',
+      body: description.body,
+      maintainer_can_modify: true,
+    });
+  });
+
+  it('Invokes octokit pull create when there are similar refs with pull requests open, but not exact refs: special character and number', async () => {
+    // setup
+    const responseCreatePullData = await import(
+      './fixtures/create-pr-response.json'
+    );
+    const responseListPullData = await import(
+      './fixtures/list-pulls-response.json'
+    );
+    const createPrResponse = {
+      headers: {},
+      status: 200,
+      url: 'http://fake-url.com',
+      data: responseCreatePullData,
+    };
+    const listPullResponse = {
+      headers: {},
+      status: 200,
+      url: 'http://fake-url.com',
+      data: responseListPullData,
+    };
+    sandbox.stub(octokit.pulls, 'list').resolves(listPullResponse);
+    const stub = sandbox
+      .stub(octokit.pulls, 'create')
+      .resolves(createPrResponse);
+
+    const similarOrigin1 = {
+      owner: 'octocat',
+      repo: 'Hello-World',
+      branch: 'new-topic-1',
+    };
+    // tests
+    await openPullRequest(octokit, upstream, similarOrigin1, description);
+    sandbox.assert.calledWithExactly(stub, {
+      owner: upstream.owner,
+      repo: similarOrigin1.repo,
+      title: description.title,
+      head: 'octocat:new-topic-1',
+      base: 'master',
+      body: description.body,
+      maintainer_can_modify: true,
+    });
+  });
+
+  it('Invokes octokit pull create when there are similar refs with pull requests open, but not exact refs: extra number', async () => {
```

Comment:
You can nest describe blocks - many of these tests share the same setup for the case of an already existing PR.

```typescript
describe('when there are similar refs with pull requests open', async () => {
  beforeEach(() => {
    // set up sinon for list pull request
  });
  it('should create another pull request if it's not the exact ref', async () => {

  });
});
```